mbed_lstools/platform_database.py

```diff
@@ -480,15 +489,101 @@ def remove(self, id, permanent=False, device_type='daplink', verbose_data=False)
 
                 return _modify_data_format(removed, verbose_data)
 
-    def update_from_web(self, url='https://os.mbed.com/api/v3/platforms/?format=json'):
+    def update_from_web(self, url=DEFAULT_UPDATE_URL):
+        """Update entries in the platform database from the API on the Mbed website.
+
+        If the update fails to contact the provided url, a `ConnectionError` will
+        be raised (part of the `requests` library). If it fails during the web
+        request or JSON parsing, a `RemotePlatformDataException` is raised.
+        If it fails during the write to the filesystem, a `DatabaseUpdateException`
+        is raised.
+
+        Returns dictionary of all updates that occurred in the following format:
+            {
+                "new": {
```

Comment:
why this function could not overwrite whole local db? isn’t this try to do same thing anyway..?

Reply:
To avoid blowing away the jlink info.